lib/jni/PrivacyGuardState.cpp

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2015-2020 Microsoft Corporation and Contributors.
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "PrivacyGuardState.hpp"
+#include "JniConvertors.hpp"
+
+namespace MAT_NS_BEGIN
+{
+
+/*static*/ CommonDataContext PrivacyGuardState::GenerateCommonDataContextObject(JNIEnv* env,
+                                                  jstring domainName,
+                                                  jstring machineName,
+                                                  jstring userName,
+                                                  jstring userAlias,
+                                                  jobjectArray ipAddresses,
+                                                  jobjectArray languageIdentifiers,
+                                                  jobjectArray machineIds,
+                                                  jobjectArray outOfScopeIdentifiers)
+{
+    CommonDataContext cdc;
+    cdc.DomainName = JStringToStdString(env, domainName);
+    cdc.MachineName = JStringToStdString(env, machineName);
+    cdc.UserName = JStringToStdString(env, userName);
+    cdc.UserAlias = JStringToStdString(env, userAlias);
+    cdc.IpAddresses = ConvertJObjectArrayToStdStringVector(env, ipAddresses);
+    cdc.LanguageIdentifiers = ConvertJObjectArrayToStdStringVector(env, languageIdentifiers);
+    cdc.MachineIds = ConvertJObjectArrayToStdStringVector(env, machineIds);
+    cdc.OutOfScopeIdentifiers = ConvertJObjectArrayToStdStringVector(env, outOfScopeIdentifiers);
+    return cdc;
+}
+
+/*static*/ std::shared_ptr<PrivacyGuard> PrivacyGuardState::getPrivacyGuardInstance()
+{
+    std::lock_guard<std::mutex> lock(m_pgInstanceGuard);
+    return m_pgInstance;
+}
+
+/*static*/ void PrivacyGuardState::setPrivacyGuardInstance(std::shared_ptr<PrivacyGuard> spPrivacyGuard)
+{
+    std::lock_guard<std::mutex> lock(m_pgInstanceGuard);
+    m_pgInstance.swap(spPrivacyGuard);
```

Comment:
Why are you swapping?

Reply:
Is it ever valid to set a PG instance if a valid (non-nullptr) one exists?

Reply:
There are instances where the instance may need to be reset. I've refactored this a bit to have an explicit 'Reset" method and to avoid having the creator perform a reset.